libs/libcds/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2019-2020 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+# if the user has not enabled libCDS, exit immediately
+if(NOT ${HPX_WITH_LIBCDS})
+  return()
+endif()
+
+include(HPX_SetupLibCDS)
+
+if(NOT libcds_POPULATED)
+  hpx_error("HPX_WITH_LIBCDS was set to ON, but HPX failed to fetch LibCDS")
+endif()
+
+hpx_add_config_define(HPX_HAVE_LIBCDS)
```

Comment:
We already generate preprocessor constants for each of the modules. Do we need this additionally?